pkg/cmd/infra/deployer/deployer.go

```diff
@@ -38,7 +47,20 @@ func NewCommandDeployer(name string) *cobra.Command {
 		Short: "Run the OpenShift deployer",
 		Long:  longCommandDesc,
 		Run: func(c *cobra.Command, args []string) {
-			if err := deploy(cfg); err != nil {
+			kClient, _, err := cfg.Config.Clients()
+			if err != nil {
+				glog.Fatal(err)
+			}
+
+			if len(cfg.DeploymentName) == 0 {
+				glog.Fatal("deployment is required")
+			}
+
+			if len(cfg.Namespace) == 0 {
```

Comment:
Also, this particular command is currently only useful as a Dockerfile entrypoint, and its arguments are provided by environment variables, so I don't see how there's a big impact at the moment. Nothing uses the CLI arg functionality here yet.